crates/primitives/src/receipt.rs

```diff
@@ -44,6 +44,40 @@ impl Receipt {
     }
 }
 
+/// A collection of receipts organized as a two-dimensional vector.
+#[derive(Clone, Debug, PartialEq, Eq, Default)]
+pub struct Receipts {
```

Comment:
We can implemented `Deref` and `DerefMut` for `Receipts` so that we can access all functions of `Vec`. If implemented you could omit `receipts.receipt_vec` from a few places

Reply:
@rakita Done